src/Sentry/Internal/BatchProcessor.cs

```diff
@@ -0,0 +1,99 @@
+using System.Timers;
+using Sentry.Protocol;
+using Sentry.Protocol.Envelopes;
+
+#if NET9_0_OR_GREATER
+using Lock = System.Threading.Lock;
+#else
+using Lock = object;
+#endif
+
+namespace Sentry.Internal;
+
+/// <summary>
+/// The Sentry Batch Processor.
+/// This implementation is not complete yet.
+/// Also, the specification is still work in progress.
+/// </summary>
+/// <remarks>
+/// Sentry Specification: <see href="https://develop.sentry.dev/sdk/telemetry/spans/batch-processor/"/>.
+/// OpenTelemetry spec: <see href="https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/batchprocessor/README.md"/>.
+/// </remarks>
+internal sealed class BatchProcessor : IDisposable
+{
+    private readonly IHub _hub;
+    private readonly BatchProcessorTimer _timer;
+    private readonly BatchBuffer<SentryLog> _logs;
+    private readonly Lock _lock;
```

Comment:
If we do want to control concurrency in this class (rather than using one of the build in data structures supporting concurrency), consider using something like `ReaderWriterLockSlim`... this is mainly useful when you have lots of reads and a few writes. However if most of the operations are writes then it doesn't buy us much/anything.

Reply:
I did indeed start with a more elaborate locking-strategy with `ReaderWriterLockSlim`.
And multi-targeted with ReaderWriterLock` on `NETFRAMEWORK` where `Thread.Abort` is allowed, since this code path may be invoked from user code.

Since we do expect more writes than reads, I then simplified it to just a "full-lock-all" approach for now, with the option to optimize later.

I just did a super quick benchmark: A "flushing" call of `EnqueueCore(SentryLog)` with a capacity of `100` takes between 2 and 3 milliseconds.

I'll do a bit of experimenting and benchmarking over the weekend with different locking strategies.